tests/test_algorithms.py

```diff
@@ -1,46 +1,139 @@
+import Orchestrator
+import pytest
```

Comment:
the name "test_algorithms" tells me very little of what this is about. I suggest you rename this file to the name of the function you are testing (I assume you are only testing the traversed cell function)